alacritty_terminal/src/tty/windows/conpty.rs

```diff
@@ -230,6 +241,35 @@ pub fn new(config: &Options, window_size: WindowSize) -> Option<Pty> {
     Some(Pty::new(conpty, conout, conin, child_watcher))
 }
 
+fn convert_custom_env(custom_env: &HashMap<String, String>) -> Option<Vec<u16>> {
+    let mut result = Vec::new();
+    if custom_env.is_empty() {
+        return None;
+    }
+
+    let mut all_env_keys = HashSet::new();
+    for (custom_key, custom_value) in custom_env {
+        let custom_key = OsStr::new(custom_key);
+        if all_env_keys.insert(custom_key.to_ascii_uppercase()) {
+            result.extend(custom_key.encode_wide());
+            result.push('=' as u16);
+            result.extend(OsStr::new(custom_value).encode_wide());
+            result.push(0);
+        }
+    }
+    for (inherited_key, inherited_value) in std::env::vars_os() {
+        if all_env_keys.insert(inherited_key.to_ascii_uppercase()) {
+            result.extend(inherited_key.encode_wide());
+            result.push('=' as u16);
+            result.extend(inherited_value.encode_wide());
+            result.push(0);
+        }
+    }
+    result.push(0);
```

Comment:
Document why it's done like that. Like it's not really clear why you need `0`, `0` at the end.